wire/msgtx.go

```diff
@@ -554,9 +657,84 @@ func (msg *MsgTx) Copy() *MsgTx {
 	return &newTx
 }
 
+// writeTxInScriptsToMsgTx allocates the memory for variable length fields in a
+// MsgTx TxIns as a contiguous chunk of memory, then fills in these fields for the
+// MsgTx by copying to a contiguous piece of memory and setting the pointer.
+func writeTxInScriptsToMsgTx(msg *MsgTx, totalScriptSize uint64) {
+	// Create a single allocation to house all of the scripts and set each
+	// input signature script to the appropriate subslice of the overall
+	// contiguous buffer.  Then, return each individual script buffer back
+	// to the pool so they can be reused for future deserializations.  This
+	// is done because it significantly reduces the number of allocations
+	// the garbage collector needs to track, which in turn improves
+	// performance and drastically reduces the amount of runtime overhead
+	// that would otherwise be needed to keep track of millions of small
+	// allocations.
+	//
+	// NOTE: It is no longer valid to call the returnScriptBuffers closure
```

Comment:
This part of the comment should be on the function definition now since it was split out.